### PR TITLE
[MM-28383] Fix nil pointer dereference in addTeamMember

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes a nil pointer dereference in addTeamMember in case of an invalid JSON request body.

#### Ticket

https://mattermost.atlassian.net/browse/MM-28383

### CODE REVIEW

File: model/team_member.go [hunk 1]
```diff
@@ -69,10 +69,12 @@ func (o *TeamUnread) ToJson() string {
 	return string(b)
 }
 
-func TeamMemberFromJson(data io.Reader) *TeamMember {
+func TeamMemberFromJson(data io.Reader) (*TeamMember, error) {
```
Comment (streamer45) [inline model/team_member.go:72]: Do we consider this a *breaking change* that should only go in a major release?
I've tried to do the *right* thing here and avoid ignoring the error but I was left with this doubt.

Comment (streamer45) [reply model/team_member.go:72]: @jespino 

Do you think it makes sense to add a `TeamMemberFromJSON` (notice the uppercase at the end) version that handles the error and start the deprecation of `TeamMemberFromJson`? Or were you suggesting to just add an internal (unexported) method?

Comment (streamer45) [reply model/team_member.go]: Alright, keeping it simple then.

**Review by jespino — APPROVED** [review]
Lgtm

**Review by mgdelacroix — APPROVED** [review]

**Comment by streamer45** [comment]
@lindalumitchell 

PR adds some error handling to fix a possible panic. I've added a unit test to cover this case.

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**